cpp/include/cuspatial/utility/device_atomics.cuh

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright (c) 2022, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace cuspatial {
+namespace detail {
+
+__device__ double atomicMin(double* addr, double val)
```

Comment:
All four of these operators should probably be implemented as specializations of a single templated function. It'll need ~5 template parameters (the value type, the bit view type, the two conversion functions between value and view types, and the actual operation to perform e.g. std::max), which is kind of gross, but it'll significantly reduce duplication and the pain of adding new operators if you need other atomics in the future.

Reply:
Let's punt this to a future PR.